src/service/viewer-impl.js

```diff
@@ -276,6 +276,16 @@ export class Viewer {
     this.performanceTracking_ = this.params_['csi'] === '1';
     dev.fine(TAG_, '- performanceTracking:', this.performanceTracking_);
 
+    /**
+     * Whether the AMP document is embedded in a viewer, such as an iframe or
+     * a web view.
+     * Notice that the value of the "origin" parameter itself is never consulted
+     * here, but instead confirmed via handshake in the `setMessageDeliverer`
+     * method.
+     * @private @const {boolean}
+     */
+    this.isEmbedded_ = this.isIframed_ || !!this.params_['origin'];
```

Comment:
I'll just add a new parameter then. We discussed with @ericfs that it might be better anyway.